Compilation from source on macOS fails in nghttp2

`make USE_BINARYBUILDER=0 VERBOSE=1 CFLAGS='-Wno-implicit-function-declaration' USE_SYSTEM_ZLIB=1` leads to:

```
cd scratch/nghttp2-1.41.0/ && \
	/private/tmp/julia/deps/srccache/nghttp2-1.41.0//configure --prefix=/private/tmp/julia/usr --build=arm64-apple-darwin21.4.0 --libdir=/private/tmp/julia/usr/lib --bindir=/private/tmp/julia/usr/tools  LDFLAGS=" -Wl,-rpath,'@loader_path/'" F77="gfortran -mmacosx-version-min=11.0 -Wa,-q" CC="clang -mmacosx-version-min=11.0 -integrated-as" CXX="clang++ -mmacosx-version-min=11.0 -integrated-as" LD="ld" --enable-lib-only
checking for gcc... clang -mmacosx-version-min=11.0 -integrated-as
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether clang -mmacosx-version-min=11.0 -integrated-as accepts -g... yes
checking for clang -mmacosx-version-min=11.0 -integrated-as option to accept ISO C89... none needed
checking whether clang -mmacosx-version-min=11.0 -integrated-as understands -c and -o together... yes
checking how to run the C preprocessor... clang -mmacosx-version-min=11.0 -integrated-as -E
checking for grep that handles long lines and -e... /opt/homebrew/bin/ggrep
checking for egrep... /opt/homebrew/bin/ggrep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking minix/config.h usability... no
checking minix/config.h presence... no
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking build system type... Invalid configuration `arm64-apple-darwin21.4.0': machine `arm64-apple' not recognized
configure: error: /bin/sh /private/tmp/julia/deps/srccache/nghttp2-1.41.0/./config.sub arm64-apple-darwin21.4.0 failed
make[1]: *** [scratch/nghttp2-1.41.0/build-configured] Error 1
make: *** [julia-deps] Error 2
```

This version of nghttp2 is too old and its configure files (config.guess, config.sub) do not know about Apple ARM.

--- a/deps/nghttp2.mk
+++ b/deps/nghttp2.mk
@@ -8,6 +8,7 @@ $(SRCCACHE)/nghttp2-$(NGHTTP2_VER).tar.bz2: | $(SRCCACHE)
 $(SRCCACHE)/nghttp2-$(NGHTTP2_VER)/source-extracted: $(SRCCACHE)/nghttp2-$(NGHTTP2_VER).tar.bz2
 	$(JLCHECKSUM) $<
 	cd $(dir $<) && $(TAR) -jxf $<
+	cp $(SRCDIR)/patches/config.sub $(SRCCACHE)/nghttp2-$(NGHTTP2_VER)/config.sub
 	touch -c $(SRCCACHE)/nghttp2-$(NGHTTP2_VER)/configure # old target
 	echo 1 > $@
 

